animation_nodes/nodes/matrix/distribute_matrices.pyx

```diff
@@ -276,16 +286,19 @@ class DistributeMatricesNode(bpy.types.Node, AnimationNode):
         iSin = sin(startAngle)
         stepCos = cos(angleStep)
         stepSin = sin(angleStep)
+        
+        Zoffset = altitude / 2 * int(self.centerAxis)
 
         for i in range(amount):
             f = <float>i * factor
 
             size = f * (endSize - startSize) + startSize
-            radius = f * (endRadius - startRadius) + startRadius
+            radius = (endRadius - startRadius) * interpolation_1(max(min(f, 1.0), 0.0)) + startRadius
```

Comment:
The logic doesn't seem to be correct here. If the end and start radius were equal, the expression would evaluate to zero, and you will end up with just `startRadius`, which I suppose is not what you intended. Can you explain the intended logic?

Also `f`  is guaranteed to be in the `[0, 1]` range, so you don't need the min and max.

Reply:
i just wanted to modify the interval between each turn in XY plane, so i mutiply the` interpolation_1(f)` , If the end and start radius were equal, means interal equal zero, in this case interpolation lose effect.

Reply:
I am not sure about that behavior. When I tried it the first time, I expected something different and it seemed to me like it didn't work. Users might feel the same way. Maybe the interpolation should be over the whole expression after the addition.
What do you think @Gerstmann-Bradley?

Reply:
I need to think more about this, will get back to this later.

Reply:
I have tested build, since I don't have a build by Omar upon this issue, I can't get a comparison about which is better.
I can only tell this curvature interpolation looks normal to me, while I think Omar's logic is right.
If no one is pointing out, I probably already think we are interpolate over the whole expression.
I am satisfied with the current result because I can't tell the difference, but I think I agree with Omar unless I find issues otherwise.

Reply:
@OmarEmaraDev you means code like this `radius = ((endRadius - startRadius) + startRadius) * radiusInterpolation.evaluate(f)`, is it right? but i tried this, the startRadius lose effect.

![spiral 3](https://user-images.githubusercontent.com/64368228/103847141-863a6a80-50da-11eb-9738-de82a5059909.gif)

Reply:
I have just come up with a way to build spiral using AN from scratch. I'm getting the same answer as @kdsin23. I am not sure the logic behind his coding, but the result matches my expectation. I have no idea now, I will leave the discussion to you two.
![example](https://user-images.githubusercontent.com/55994236/103848790-794e5480-50a3-11eb-9458-7e096caa440d.png)

Reply:
I idea was to let the base case be a constant interpolation at value 1. But I guess this one might be confusing as well as evident by this discussion. A base linear interpolation might be more familiar to users. So just leave it like you had originally for now. We will make sure to document those cases properly.

Reply:
ok, i will do that